src/schemables/WithJson/instances/eq.ts

```diff
@@ -0,0 +1,34 @@
+/**
+ * A basal schemable for Json and JsonString
+ *
+ * @since 1.0.2
+ */
+import * as J from 'fp-ts/Json'
+import * as RA from 'fp-ts/ReadonlyArray'
+import * as RR from 'fp-ts/ReadonlyRecord'
+
+import * as Eq_ from '../../../base/EqBase'
+import { WithJson1 } from '../definition'
+
+/**
+ * @since 1.0.2
+ * @category Instances
+ */
+export const Eq: WithJson1<Eq_.URI> = {
+  json: {
+    equals: (x, y) =>
+      x === y ||
+      (Array.isArray(x) &&
+        Array.isArray(y) &&
+        x.length === y.length &&
+        RA.getEq<J.Json>(Eq.json).equals(x, y)) ||
+      (typeof x === 'object' &&
+        typeof y === 'object' &&
+        x !== null &&
+        y !== null &&
+        !Array.isArray(x) &&
+        !Array.isArray(y) &&
+        RR.getEq<string, J.Json>(Eq.json).equals(x as J.JsonRecord, y as J.JsonRecord)),
+  },
+  jsonString: Eq_.string,
```

Comment:
I think this needs to be fixed, two JsonStrings are technically equal if they are equal in their parsed states. So this can pretty simply be a parse x, parse y, then `Eq.json(parsedX, parsedY)`

Reply:
oh yeah, good point. key order is not guaranteed.

or.. it is. but by insertion order; `JSON.stringify({ a: 1, b: 2 }) !== JSON.stringify({ b: 2, a: 1 })`
not very useful for equality checks